subcommands/common_config.go

```diff
@@ -220,22 +225,62 @@ func SetUpdatesConfig(opts *SetUpdatesConfigOptions, reportedTag string, reporte
 	}
 
 	changed := false
-	if opts.UpdateApps != "" && configuredApps != opts.UpdateApps {
-		if strings.TrimSpace(opts.UpdateApps) == "," {
-			opts.UpdateApps = ""
+	if opts.UpdateAppsGiven || opts.InheritApps {
+
+		if opts.UpdateTagGiven && opts.UpdateTag != "" {
+			configuredTag = opts.UpdateTag
+		}
+		if opts.InheritApps && sota.Has("pacman.compose_apps") {
+			fmt.Printf("Changing apps from: [%s] -> inherit from parent\n", configuredApps)
+			DieNotNil(sota.Delete("pacman.docker_apps"))
+			DieNotNil(sota.Delete("pacman.compose_apps"))
+			changed = true
+		} else if configuredApps != opts.UpdateApps && opts.UpdateAppsGiven {
+			if strings.TrimSpace(opts.UpdateApps) == "," {
+				targets, err := api.TargetsList(factory)
+				DieNotNil(err)
+				version := 0
+				var final_apps []string
+				for _, target := range targets {
+					custom, err := api.TargetCustom(target)
+					DieNotNil(err)
+					set := make(map[string]bool)
+					var apps []string
+					if slices.Contains(custom.Tags, configuredTag) {
+						for app := range custom.ComposeApps {
+							if _, ok := set[app]; !ok {
+								apps = append(apps, app)
+							}
+						}
+						if v, err := strconv.Atoi(custom.Version); err == nil {
+							if v > version {
+								final_apps = apps
+								version = v
+							}
+						}
+					}
+				}
+				opts.UpdateApps = strings.Join(final_apps, ",")
+			}
+			fmt.Printf("Changing apps from: [%s] -> [%s]\n", configuredApps, opts.UpdateApps)
+			sota.Set("pacman.docker_apps", opts.UpdateApps)
+			sota.Set("pacman.compose_apps", opts.UpdateApps)
+			changed = true
 		}
-		fmt.Printf("Changing apps from: [%s] -> [%s]\n", configuredApps, opts.UpdateApps)
-		sota.Set("pacman.docker_apps", opts.UpdateApps)
-		sota.Set("pacman.compose_apps", opts.UpdateApps)
-		changed = true
 	}
-	if opts.UpdateTag != "" && configuredTag != opts.UpdateTag {
-		if strings.TrimSpace(opts.UpdateTag) == "," {
-			opts.UpdateTag = ""
+	if opts.UpdateTagGiven || opts.InheritTag {
+		if opts.InheritTag && sota.Has("pacman.tags") {
+			fmt.Printf("Changing tag from: [%s] -> inherit from parent\n", configuredTag)
+			DieNotNil(sota.Delete("pacman.tags"))
+			changed = true
+		} else if configuredTag != opts.UpdateTag {
+			if strings.TrimSpace(opts.UpdateTag) == "," {
+				opts.UpdateTag = ""
+			}
```

Comment:
> I just copied this logic from existing implementation. I can get rid of it through this PR no worries.

After I wrote a comment, I had a feeling that you copied it _as is_ and I understood it wrong.
This is not your fault, it is the complexity of _all variants we need to support_ that drives seeds the confusion.

Reply:
@doanac do we want to continue supporting the "unset tag" option?
I _believe_ we don't want to allow devices with no tag.

@StealthyCoder if Andy gives a green light to this, please, make it as a separate commit in this PR, explaining the rationale.

Reply:
Its safe to "unset" because a `sota.toml` will have a tag - lmp-device-register forces that. This could be handy for something like:
* A device in a device-group is misbehaving
* Move device to a "debug" type tag
* After debugging, "unset" to get it back to device-group.